client/txn.go

```diff
@@ -90,6 +90,11 @@ func newTxn(db DB, depth int) *Txn {
 	return txn
 }
 
+// GetState returns the transaction protobuf.
+func (txn *Txn) GetState() proto.Transaction {
```

Comment:
haveTxnWrite gets reflected in proto.Transaction.Writing by the txn coordinator. haveEndTxn is only used to prevent a txn from sending another end transaction message. I suppose we could send a notification back through the proto message that the transaction has indeed already ended, but I see no use for it. Perhaps when I add the DB.TxnRetry interface later to allow us to run retryable transaction routines, we will need it to decide when to auto commit. That's a separate change.